src/util/time.cpp

```diff
@@ -6,3 +6,33 @@ LLTIMER Time::s_timer;
 bool Time::s_testMode = false;
 // static
 qint64 Time::s_testElapsed_nsecs = 0;
+
+// static
+QString Time::formatSeconds(double dSeconds, bool showCentis) {
+    if (dSeconds < 0) {
+        return "?";
+    }
+
+    const int days = static_cast<int>(dSeconds) / kSecondsPerDay;
+    dSeconds -= days * kSecondsPerDay;
+
+    // NOTE(uklotzde): Time() constructs a 'null' object, but
+    // we need 'zero' here.
+    QTime t = QTime(0, 0).addMSecs(dSeconds * kMillisPerSecond);
+
+    QString formatString =
+            (days > 0 ? (QString::number(days) %
+                         QLatin1String("'d', ")) : QString()) %
+            QLatin1String(days > 0 || t.hour() > 0 ? "hh:mm:ss" : "mm:ss") %
+            QLatin1String(showCentis ? ".zzz" : "");
```

Comment:
Unrelated, but I have just noticed it: 
http://stackoverflow.com/questions/31477309/timestamp-in-iso-8601-the-last-6-digits-yyyy-mm-ddthhmmss
it should be QLocale::decimalPoint()